contracts/zkdvrf.sol

```diff
@@ -172,15 +173,17 @@ contract zkdvrf is Ownable {
     // accept a set of partial evals
     // take sigma as a param, basically a point that the operator submits (combination of subset of partial evals)
     // take the gpk as stored in contract
-    function generateRandom(Pairing.G1Point memory sigma) public onlyOwner{
-        require(roundToRandom[currentRoundNum] == bytes32(0), "Answer for round already exists");
+    function submitRandom(IPseudoRand.PseudoRandom memory pseudo) public onlyOwner{
+        require(roundToRandom[currentRoundNum].value == bytes32(0), "Answer for round already exists");
         require(roundSubmissionCount[currentRoundNum] >= threshold, "Partial evaluation threshold not reached");
-        require(IPseudoRand(pseudoRand).verifyPseudoRand(bytes(roundInput[currentRoundNum]), sigma, gpkVal), "Incorrect random submitted");
-        roundToRandom[currentRoundNum] = keccak256(abi.encodePacked(sigma.X, sigma.Y));
+        require(IPseudoRand(pseudoRand).verifyPseudoRand(bytes(roundInput[currentRoundNum]), pseudo.proof, gpkVal), "Incorrect random submitted");
+        bytes32 value = keccak256(abi.encodePacked(pseudo.proof.x, pseudo.proof.y));
+        require(pseudo.value == value, "Incorrect pseudorandom value");
```

Comment:
why do we want to take this as an user input and compare? instead of assigning the value directly. (gas savings)

Reply:
It is actually useful to avoid bugs/inconsistence. I just found out that the keccak hash in rust produces different value than the one in the contract due to some data format difference.